framework/test/unit/modules/nft/endpoint.spec.ts

```diff
@@ -360,80 +360,124 @@ describe('NFTEndpoint', () => {
+				supportedCollectionIDs: ['*'],
+			});
+		});
 
-			await expect(endpoint.getCollectionIDs(maxLengthContext)).rejects.toThrow(
-				`'.chainID' must NOT have more than 8 characters`,
-			);
+		it('should return an empty array when supportedNFTsStoreData is empty', async () => {
+			const context = createTransientModuleEndpointContext({
+				stateStore,
+			});
+
+			await expect(endpoint.getSupportedCollectionIDs(context)).resolves.toEqual({
+				supportedCollectionIDs: [],
+			});
 		});
 
-		it('should return empty list if provided chainID does not exist', async () => {
+		it('should return a supportedCollectionIDs array containing the chainID + "********" when the `supportedCollectionIDArray` property is empty', async () => {
+			const chainID = utils.getRandomBytes(LENGTH_CHAIN_ID);
+
+			await supportedNFTsStore.save(methodContext, chainID, {
+				supportedCollectionIDArray: [],
+			});
+
 			const context = createTransientModuleEndpointContext({
 				stateStore,
-				params: {
-					chainID: utils.getRandomBytes(LENGTH_CHAIN_ID).toString('hex'),
-				},
 			});
 
-			await expect(endpoint.getCollectionIDs(context)).resolves.toEqual({ collectionIDs: [] });
+			await expect(endpoint.getSupportedCollectionIDs(context)).resolves.toEqual({
+				supportedCollectionIDs: [`${chainID.toString('hex')}********`],
+			});
 		});
 
-		it('should return supported collections of the provided chain', async () => {
+		it('should return an array with supported collection IDs', async () => {
 			const chainID = utils.getRandomBytes(LENGTH_CHAIN_ID);
+			const collectionID = utils.getRandomBytes(LENGTH_COLLECTION_ID);
 
-			const supportedCollections = [
-				{
-					collectionID: utils.getRandomBytes(LENGTH_COLLECTION_ID),
-				},
-				{
-					collectionID: utils.getRandomBytes(LENGTH_COLLECTION_ID),
-				},
-			];
+			await supportedNFTsStore.save(methodContext, chainID, {
+				supportedCollectionIDArray: [
+					{
+						collectionID,
+					},
+				],
+			});
+
+			const context = createTransientModuleEndpointContext({
+				stateStore,
+			});
+
+			await expect(endpoint.getSupportedCollectionIDs(context)).resolves.toEqual({
+				supportedCollectionIDs: [Buffer.concat([chainID, collectionID]).toString('hex')],
+			});
+		});
+
+		it('should return an array with supported collection IDs for native chains when chain is equal to ownChainID', async () => {
+			const chainID = ownChainID;
+			const collectionID = utils.getRandomBytes(LENGTH_COLLECTION_ID);
+			await supportedNFTsStore.save(methodContext, chainID, {
+				supportedCollectionIDArray: [
+					{
+						collectionID,
+					},
+				],
+			});
+			const context = createTransientModuleEndpointContext({
+				stateStore,
+			});
+			await expect(endpoint.getSupportedCollectionIDs(context)).resolves.toEqual({
+				supportedCollectionIDs: [Buffer.concat([chainID, collectionID]).toString('hex')],
+			});
+		});
+
+		it('should return an array of unique supported native collection IDs', async () => {
```

Comment:
As per my previous comment I still coudn't understand how this test checks `native collection IDs` when the owner of the nft's saved to the NFT sstore is an address not the `ownChainID`. Please look at the logic b/w lines L187-L191 in endpoint.ts